### PR TITLE
CI/TST: Error on PytestUnraisableExceptionWarning instead of using psutil to check open resources

### PR DESCRIPTION
**Pull request by mroeschke**
Using `psutil` to check for open resources in tests has been a false positive for a while, picking up unrelated connections such as the database services.

Erroring on `ResourceWarnings`, captured by `PytestUnraisableExceptionWarning`, should be sufficient for validating that resources are not left open as a result of a test. https://github.com/pytest-dev/pytest/issues/9825

### CODE REVIEW

**Comment by jbrockmendel** [comment]
there have been times in the past where we _did_ leave files open incorrectly, which I think the check this removes would catch.  does the check this adds still catch those?

**Comment by mroeschke** [comment]
Correct for example

```
 % git diff
diff --git a/pandas/tests/frame/test_api.py b/pandas/tests/frame/test_api.py
index e5787a7f16..c280fb9748 100644
--- a/pandas/tests/frame/test_api.py
+++ b/pandas/tests/frame/test_api.py
@@ -379,3 +379,7 @@ class TestDataFrameMisc:
         df = DataFrame()
         with tm.assert_produces_warning(None):
             inspect.getmembers(df)
+
+
+def test_open():
+    open("/dev/null")

% pytest pandas/tests/frame/test_api.py

...

pandas/tests/frame/test_api.py .................................F

...
    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E
E               Traceback (most recent call last):
E                 File "...pandas/tests/frame/test_api.py", line 385, in test_open
E                   open("/dev/null")
E               ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='r' encoding='UTF-8'>

../opt/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarnin
```

**Comment by jbrockmendel** [comment]
awesome, thanks for explaining

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -1617,11 +1617,3 @@ def __exit__(
         traceback: TracebackType | None,
     ) -> None:
         self.close()
-
-    def __del__(self) -> None:
```
Comment (phofl) [inline pandas/io/excel/_base.py:1621]: Why did you delete this?

Comment (mroeschke) [reply pandas/io/excel/_base.py:1621]: IMO we shouldn't rely on garbage collection to perform a final closure of files from `ExcelWriter`. It didn't work correctly from the `ResourceWarning`s issues picked up from this PR, and would be a band-aid fix at best

Comment (phofl) [reply pandas/io/excel/_base.py:1621]: thx

File: pandas/tests/scalar/timestamp/test_timestamp.py [hunk 2]
```diff
@@ -152,6 +156,11 @@ def test_names(self, data, time_locale):
 
     def test_is_leap_year(self, tz_naive_fixture):
         tz = tz_naive_fixture
+        if not IS64 and tz == tzlocal():
+            # https://github.com/dateutil/dateutil/issues/197
+            pytest.skip(
+                "tzlocal() on a 32 bit platform causes internal overflow errors"
+            )
```
Comment (jbrockmendel) [inline pandas/tests/scalar/timestamp/test_timestamp.py:163]: are we giving incorrect results in these cases?

File: pandas/tests/io/excel/conftest.py [hunk 3]
```diff
@@ -42,26 +39,3 @@ def read_ext(request):
...
-    # GH#30162
-    psutil = td.safe_import("psutil")
-    if not psutil:
-        yield
-
-    else:
-        proc = psutil.Process()
-        flist = proc.open_files()
-        yield
-        flist2 = proc.open_files()
```
Comment (jbrockmendel) [inline pandas/tests/io/excel/conftest.py:66]: documenting in case we wind up keeping this: looks like all of the flaky failures have raddr.port of either 5432 or 3306 (mostly 5432), which correspond to postgres and mysql defaults, respectively.  We could filter those out here

File: pandas/tests/scalar/timestamp/test_timestamp.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/scalar/timestamp/test_timestamp.py:163]: Running the 32 bit container locally it appears this still returns the correct result but hits the traceback in the dateutil issue

```
     File "pandas/_libs/tslibs/tzconversion.pyx", line 125, in pandas._libs.tslibs.tzconversion.Localizer.utc_val_to_local_val
      return utc_val + _tz_localize_using_tzinfo_api(
    File "pandas/_libs/tslibs/tzconversion.pyx", line 736, in pandas._libs.tslibs.tzconversion._tz_localize_using_tzinfo_api
      dt = _astimezone(dts, tz)
    File "pandas/_libs/tslibs/tzconversion.pyx", line 768, in pandas._libs.tslibs.tzconversion._astimezone
      return tz.fromutc(result)
    File "/root/virtualenvs/pandas-dev/lib/python3.8/site-packages/dateutil/tz/_common.py", line 144, in fromutc
      return f(self, dt)
    File "/root/virtualenvs/pandas-dev/lib/python3.8/site-packages/dateutil/tz/_common.py", line 261, in fromutc
      _fold = self._fold_status(dt, dt_wall)
    File "/root/virtualenvs/pandas-dev/lib/python3.8/site-packages/dateutil/tz/_common.py", line 196, in _fold_status
      if self.is_ambiguous(dt_wall):
    File "/root/virtualenvs/pandas-dev/lib/python3.8/site-packages/dateutil/tz/tz.py", line 254, in is_ambiguous
      naive_dst = self._naive_is_dst(dt)
    File "/root/virtualenvs/pandas-dev/lib/python3.8/site-packages/dateutil/tz/tz.py", line 260, in _naive_is_dst
      return time.localtime(timestamp + time.timezone).tm_isdst
  OverflowError: timestamp out of range for platform time_t
```

File: pandas/tests/io/excel/conftest.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/io/excel/conftest.py:66]: FWIW I have seen the port 443 show up a few times too

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
merging, I guess we should backport to get rid of these on the 2.0.x branch?

**Comment by phofl** [comment]
thx @mroeschke

**Comment by jbrockmendel** [comment]
nice!
